pkgs/by-name/ts/tsukimi/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  lib,
+  pkgs,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  cargo,
+  pkg-config,
+  glib,
+  mpv,
+  ffmpeg,
+  libadwaita,
+  gst_all_1,
+  gtk4,
+  openssl,
+  libepoxy,
+  gsettings-desktop-schemas,
+  wrapGAppsHook4,
+  makeDesktopItem,
+  gettext,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "tsukimi";
+  version = "0.12.2";
+
+  src = fetchFromGitHub {
+    owner = "tsukinaha";
+    repo = "tsukimi";
+    rev = "v${version}";
+    hash = "sha256-pJ+SUNGQS/kqBdOg21GgDeZThcjnB0FhgG00qLfqxYA=";
+  };
+
+  icon = fetchurl {
+    url = "https://aur.archlinux.org/cgit/aur.git/plain/tsukimi.png?h=tsukimi-git&id=22721d9f058dafeec7f07907aec68fa232ccd2c7";
+    hash = "sha256-BKcS1gu/XMiHOvJPuP92tH8n73e6RceQH7fTzSbdaJM=";
+  };
+
+  cargoHash = "sha256-PCJiSyfEgK8inzoRmRvnAU50kLnyVhNrgLrwtBUFpIU=";
+
+  nativeBuildInputs = [
+    cargo
+
+    pkg-config
+    glib
+
+    wrapGAppsHook4
+    gtk4
+  ];
+
+  buildInputs =
+    [
+      mpv
+      ffmpeg
+      libadwaita
+
+      openssl
+      libepoxy
+      glib
+
+      gsettings-desktop-schemas
+      gettext
+      gtk4
+    ]
+    ++ (with gst_all_1; [
+      gstreamer
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-bad
+      gst-plugins-ugly
+      gst-libav
+    ]);
+
+  doCheck = false;
+
+  desktopItem = makeDesktopItem {
+    name = "Tsukimi";
+    exec = "tsukimi";
+    type = "Application";
+    icon = "tsukimi";
+    categories = [ "AudioVideo" ];
+    startupWMClass = "moe.tsuna.tsukimi";
+
+    desktopName = "Tsukimi";
+  };
+
+  postPatch = ''
+    substituteInPlace build.rs \
+      --replace-fail 'i18n/locale' "$out/share/locale"
+
+    substituteInPlace src/main.rs \
+      --replace-fail '/usr/share/locale' "$out/share/locale"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/share/glib-2.0/schemas
```

Comment:
```suggestion
```

Reply:
```
       > Executing cargoInstallHook
       > install: cannot create regular file '/nix/store/b0a5a3lr34wafj95s92v7g92gmcj5ilq-tsukimi-0.12.2/share/glib-2.0/schemas/': Not a directory
```

I suppose we should keep it.

Reply:
Oh it's because you didn't use `install` correctly.